javafmllanguage/src/main/java/net/minecraftforge/fml/javafmlmod/AutomaticEventSubscriber.java

```diff
@@ -49,6 +56,10 @@ public static void inject(ModContainer mod, ModFileScanData scanData, ClassLoade
         var defaultBus = new EnumData(null, "FORGE");
 
         for (var data : targets) {
+            if (!FMLEnvironment.production && onlyIns.contains(data.clazz().getClassName())) {
+                throw new RuntimeException("Found @OnlyIn on @EventBusSubscriber class " + data.clazz().getClassName() + " - this is not allowed you should instead use the dist parameter on EventBusSubscriber annotation");
```

Comment:
```suggestion
                throw new RuntimeException("Found @OnlyIn on @EventBusSubscriber class " + data.clazz().getClassName() + " - this is not allowed as it causes crashes. Remove the OnlyIn and set value=Dist.CLIENT in the EventBusSubscriber annotation instead");
```
Replacing the "Dist.CLIENT" part in my suggestion based on the dist inside the OnlyIn.